File: java/src/main/java/gov/doc/ntia/sigmf/examples/location/GcsWithSpheroidExample.java
package gov.doc.ntia.sigmf.examples.location;

import gov.doc.ntia.sigmf.Capture;
import gov.doc.ntia.sigmf.Global;
import gov.doc.ntia.sigmf.MetaDoc;
import gov.doc.ntia.sigmf.examples.Example;
import gov.doc.ntia.sigmf.examples.ExampleUtils;
import gov.doc.ntia.sigmf.ext.global.location.Datum;
import gov.doc.ntia.sigmf.ext.global.location.GeographicCoordinateSystem;
import gov.doc.ntia.sigmf.ext.global.location.Spheroid;
import java.util.Calendar;
import java.util.Date;
import java.util.List;

public class GcsWithSpheroidExample implements Example {
  @Override
  public MetaDoc getExample() {
    Date now = Calendar.getInstance().getTime();
    MetaDoc metadoc = new MetaDoc();

    Global global = ExampleUtils.getGlobal();
    List<Capture> captures = ExampleUtils.getCaptures(now);
    metadoc.setGlobal(global);
    metadoc.setCaptures(captures);
    GeographicCoordinateSystem gcs = new GeographicCoordinateSystem();
    gcs.setId("EPSG:4326");
    Spheroid spheroid = new Spheroid();
    spheroid.setSemimajor(6378137.0);
    spheroid.setSemiminor(6356752.3142);
    spheroid.setFlattening((1 / 298.257223563));
    spheroid.setPrimeMeridian("Greenwich");
    spheroid.setMeridianOffset(0.0);
    Datum datum = new Datum();
    datum.setSpheroid(spheroid);
    gcs.setDatum(datum);
    String description = "\"World Geodetic System 1984 - Horizontal component of 3D system.";
    description += " Used by the GPS and for NATO military geodetic surveying.";
    gcs.setDescription(description);
    global.setCoordinateSystem(gcs);

    return metadoc;
  }
}

File: C#/sigmf/sigmf/examples/location/GcsWithSpheroidExample.cs
using System;
using System.Collections.Generic;
using System.Text;

namespace sigmf.examples.location
{
    class GcsWithSpheroidExample : Example
    {
        public override MetaDoc getExample()
        {
            DateTime now = getCurrentTime();
            MetaDoc metaDoc = new MetaDoc();

            Globals globals = ExampleUtils.getGlobals(now);
            List<Capture> captures = ExampleUtils.getCaptures(now);
            metaDoc.setGlobal(globals);
            metaDoc.setCaptures(captures);
            GeographicCoordinateSystem gcs = new GeographicCoordinateSystem();
            gcs.Id = "EPSG:4326";
            Datum datum = new Datum();
            Spheroid spheroid = new Spheroid();
            spheroid.Semimajor = 6378137.0;
            spheroid.Semiminor = 6356752.3142;
            spheroid.Flattening = 1 / (298.257223563);
            spheroid.PrimeMeridian = "Greenwich";
            spheroid.MeridianOffset = 0.0;
            datum.Spheroid = spheroid;
            gcs.Datum = datum;
            gcs.Description = "World Geodetic System 1984 - Horizontal component of 3D system. Used by the GPS satellite navigation system and for NATO military geodetic surveying.";

            globals.CoordinateSystem = gcs;

            return metaDoc;
        }
        private DateTime getCurrentTime()
        {
            DateTime st = new DateTime(1970, 1, 1, 0, 0, 0, DateTimeKind.Utc);
            TimeSpan t = (DateTime.Now.ToUniversalTime() - st);
            return st;
        }
    }
}
